docs/developer.md

```diff
@@ -0,0 +1,188 @@
+- change its 'image' field and point to your repo/image
+- optional (but encouraged during development): set renovate's dry run parameter
+  to `full`, in case you don't want to create PRs in the stage repositories:
+```yaml
+    env:
+    - name: LOG_LEVEL
+        value: debug
+    - name: RENOVATE_DRY_RUN
+        value: 'full'
+```
+- manually create the job, let it run and check the logs for any error
+    - if you are using the UI, you can click on the "+" button on the top right,
+    paste your yaml in the window that will appear, then clicking on 'Create'
+    - if you prefer using the cli, you can use `oc create -f <path to your yaml file>` 
+
+Creating a job in this manner will make it run immediatelly, so you don't
+need to wait until the next cronjob-invoked run. If you need to make another
+run, simply delete the job and then recreate it (demanding another run to the
+same job might not be possible from the UI):
+- to delete your job via the UI: navigate to your job's page, then under the
+  'Actions' dropdown, select 'Delete Job'
+    - if you don't have a 'Jobs' tab in your interface, you can go to Search > 
+    check 'Jobs' in the 'Resources' dropdown > select 'Name' in the filter
+    dropdown, then enter the beginning of your job name > sort results by
+    creation date > click on the latest job
+- to delete it via the cli, use `oc delete job <your job name>`
+
+## Release process
+
+In this section we will describe the steps to make a release. We will see each
+step separately in detail, and also provide a checklist by the end.
+
+### Propose and merge your changes
+
+Changes to the MintMaker project fall into one of these three categories:
+- changes to the controller code (this repo)
+- changes to the custom image used by MintMaker
+- changes to the configuration file
+
+Propose your changes as usual, get your reviews and finally merge them. In the
+first two cases, the automatic build in Konflux will be triggered when your
+changes are merged into the main branch, and the next step will be to wait for
+them to finish.
+
+If you are making routine changes for the weekly release and believe no reviews
+are necessary, you can skip the reviews (e.g., the changes are simply konflux
+reference udpates).
+
+In case you are making changes only to the configuration file, you can skip to
+the [stage run step](./developer.md#stage-run-to-check-the-new-build)
+
+
+### Wait until Konflux automatic build finishes
+
+The status of the Konflux build job can be checked in github via the 'Actions'
+tab, or in the PR checks after the merge event. The build job is called
+`mintmaker-on-pull-request`.
+
+When it is finished, Konflux will push the new image to Quay, so you can watch
+it to check when the job is done as well. Images are labelled with the git commit
+sha, so you can use that to identify which image corresponds to which commit.
+
+
+### Stage run to check the new build
```

Comment:
I would perhaps also mention that to promote to stage one needs to merge the automatically created MR in infra-deployments

Reply:
Good catch! 😁